### ISSUE TITLE
gitgutter runs diffs on git commit messages

### ISSUE THREAD

**Issue by elyscape** [airblade/vim-gitgutter#239]
This doesn't seem particularly useful and, on Windows systems without vim-proc that have realtime diffing enabled, it causes the vimrun.exe windows to pop up in the background occasionally.  It's not a huge deal but it might be nice to check and not run diffs when `&ft` is `gitcommit`.

**Comment by airblade** [comment]
We could generalise this to not running diffs on any file within the `.git/` directory.

**Comment by airblade** [comment]
Please could you try this patch?

```diff
diff --git a/autoload/gitgutter/utility.vim b/autoload/gitgutter/utility.vim
index 60d62f1..302f380 100644
--- a/autoload/gitgutter/utility.vim
+++ b/autoload/gitgutter/utility.vim
@@ -10,7 +10,11 @@ function! gitgutter#utility#warn(message)
 endfunction
 
 function! gitgutter#utility#is_active()
-  return g:gitgutter_enabled && gitgutter#utility#exists_file()
+  return g:gitgutter_enabled && gitgutter#utility#exists_file() && gitgutter#utility#not_git_dir()
+endfunction
+
+function! gitgutter#utility#not_git_dir()
+  return gitgutter#utility#directory_of_file() !~ '\.git\(/.*\)\?$'
 endfunction
 
 " A replacement for the built-in `shellescape(arg)`.
```

**Comment by elyscape** [comment]
Doesn't work on Windows when the file is in a subdirectory of the .git directory because of the path separator.  Possible solution:
```viml
  return gitgutter#utility#directory_of_file() !~ '\.git\([/\\].*\)\?$'
```

**Comment by elyscape** [comment]
Also, it might be a good idea to get the full path instead of just the path from whatever the current working directory is:
```viml
function! gitgutter#utility#not_git_dir()
  return gitgutter#utility#full_path_to_directory() !~ '\.git\([/\\].*\)\?$'
endfunction

function! gitgutter#utility#full_path_to_directory()
  return fnamemodify(s:file, ':p:h')
endfunction
```

**Comment by airblade** [comment]
The code looks good.  I didn't follow what you said about committing to a submodule – why would that involve anything in a subdirectory of `.git/`?

**Comment by elyscape** [comment]
For submodules, the .git directory is actually a symlink to `.git/modules/path/to/submodule` in the primary repository.  Normally this isn't an issue, but, on systems with no or less-robust support for symlinks (i.e. Windows), git instead puts a text file in place with the destination of the symlink as the contents.  You can enable this behavior with systems with proper symlink support but setting `core.symlinks` to `false`.  When committing to a submodule with this behavior active, the path to `COMMIT_EDITMSG` will be `repo/.git/modules/path/to/submodule/COMMIT_EDITMSG` instead of `repo/path/to/submodule/.git/COMMIT_EDITMSG`.

**Comment by airblade** [comment]
Thanks for the explanation – it all makes sense now.

I've merged the change in 8345c35.
